src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -553,13 +561,10 @@ static UCS_CLASS_INIT_FUNC(uct_cuda_ipc_iface_t, uct_md_h md, uct_worker_h worke
         return status;
     }
 
-    self->config.max_poll            = config->max_poll;
-    self->config.max_streams         = config->max_streams;
-    self->config.enable_cache        = config->enable_cache;
-    self->config.enable_get_zcopy    = config->enable_get_zcopy;
-    self->config.max_cuda_ipc_events = config->max_cuda_ipc_events;
-    self->config.bandwidth           = UCS_CONFIG_DBL_IS_AUTO(config->bandwidth) ?
-                                       uct_cuda_ipc_iface_get_bw() : config->bandwidth;
+    self->config           = config->self;
+    self->config.bandwidth = UCS_CONFIG_DBL_IS_AUTO(config->self.bandwidth) ?
+                                    uct_cuda_ipc_iface_get_bw() :
+                                    config->self.bandwidth;
```

Comment:
seems now you do not need to reinitialize. I.e.
```suggestion
    self->config           = config->self;
    if (UCS_CONFIG_DBL_IS_AUTO(config->self.bandwidth)) {
        self->config.bandwidth = uct_cuda_ipc_iface_get_bw() ;
    }
```